index.js

```diff
@@ -17,50 +17,96 @@ const QUESTIONS = [{
 const SUBMISSION = [
 
 ]
-
+//adding middleware (are sort of inbuilt functions that give permissions to use certain expressJS features in this case) to get data from post method to a js object
+app.use(express.urlencoded({extended : true}));
+
+//function that checks if the user is existing and the password is the same:
+const authenticateUser = (email,password)=>{
+  for(user of USERS){
+    if(user.email === email && user.password === password){
+      return true;
+    }
+  }
+  return false;
+};
+
+//function that check if the user is existing:
+const existingUser = (email) =>{
+  for(user of USERS){
+    if(user.email === email){
+      return true;
+    }
+  }
+  return false;
+}
 app.post('/signup', function(req, res) {
-  // Add logic to decode body
-  // body should have email and password
 
+  // Add logic to decode body
+  const {email, password} = req.body;
 
+  // body should have email and password
   //Store email and password (as is for now) in the USERS array above (only if the user with the given email doesnt exist)
 
-
+  if(!existingUser(email)){
+    USERS.push(req.body);
+  }
   // return back 200 status code to the client
-  res.send('Hello World!')
-})
+  res.sendStatus(200);
+});
 
 app.post('/login', function(req, res) {
-  // Add logic to decode body
   // body should have email and password
-
-  // Check if the user with the given email exists in the USERS array
-  // Also ensure that the password is the same
-
-
-  // If the password is the same, return back 200 status code to the client
+  const {email, password} = req.body;
+  // Checking if the user with the given email exists in the USERS array
+  //if the user dosennt exist we redirect to the login page
+  if(!existingUser){
+    res.redirect("/signup");
+  }
+  // authenticating password: 
+  if(authenticateUser(email,password)){
   // Also send back a token (any random string will do for now)
+  //token(cookie) is sent below
+    res.cookie("token","myTokenName",{
+      httpOnly: true,
+      expires: new Date(Date.now() + 60*2000)
+    });
+    // If the password is the same, return back 200 status code to the client
+    res.sendStatus(200);
```

Comment:
you need to add return here otherwise last `sendStatus(401)` below will also get executed.